test/e2e/e2eslow/self_hosted_test.go

```diff
@@ -92,6 +93,40 @@ func testCreateSelfHostedClusterWithBootMember(t *testing.T) {
 	}
 }
 
+func testCreateSelfHostedClusterWithBootMemberNoAutoRemoval(t *testing.T) {
+	f := framework.Global
+
+	bootEtcdPod, err := startEtcd(f)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer f.KubeClient.CoreV1().Pods(f.Namespace).Delete(bootEtcdPod.Name, metav1.NewDeleteOptions(0))
+
+	bootURL := fmt.Sprintf("http://%s:2379", bootEtcdPod.Status.PodIP)
+
+	t.Logf("boot etcd URL: %s", bootURL)
+
+	c := e2eutil.NewCluster("test-etcd-", 1)
+	c = e2eutil.ClusterWithSelfHosted(c, &spec.SelfHostedPolicy{
+		BootMemberClientEndpoint: bootURL,
+		SkipBootMemberRemoval:    true,
+	})
+	testEtcd, err := e2eutil.CreateCluster(t, f.KubeClient, f.Namespace, c)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer func() {
+		if err := e2eutil.DeleteCluster(t, f.KubeClient, testEtcd); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	if _, err := e2eutil.WaitUntilSizeReached(t, f.KubeClient, 2, 120*time.Second, testEtcd); err != nil {
```

Comment:
This won't succeed.
There is an assumption that the boot member needs to be removed before scaling.